flutter_app/lib/widgets/waypoint_queue_widget.dart

```diff
@@ -0,0 +1,171 @@
+    widget.queueWaypoints.sendWaypointWithoutQueue(
+        widget.systemId, widget.componentId, _latitude, _longitude, _altitude);
+  }
+
+  /// Sends the first waypoint in the queue to the drone.
+  void _sendNextWaypointInQueue() {
+    widget.queueWaypoints.sendNextWaypointInQueue();
+    setState(() {
+      _widgetKey.currentState?.reassemble();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        const Text(
+          'Waypoint Queue',
+          style: TextStyle(fontWeight: FontWeight.bold),
+        ),
+        DataTable(
+          columns: const [
+            DataColumn(label: Text('Latitude')),
+            DataColumn(label: Text('Longitude')),
+            DataColumn(label: Text('Altitude')),
+          ],
+          rows: widget.queueWaypoints.waypointQueue
+              .map(
+                ((element) => DataRow(
+                      cells: <DataCell>[
+                        DataCell(Text(element.x.toString())),
+                        DataCell(Text(element.y.toString())),
+                        DataCell(Text(element.z.toString())),
+                      ],
+                    )),
+              )
+              .toList(),
+        ),
+        const SizedBox(height: 16),
+        ElevatedButton(
+            onPressed: _sendNextWaypointInQueue,
+            child: const Text('Send Next Waypoint in Queue to Drone')),
+        const SizedBox(height: 16),
+        const Text(
+          'Enter a Waypoint Below',
+          style: TextStyle(fontWeight: FontWeight.bold),
+        ),
+        Padding(
+          padding: const EdgeInsets.all(10.0),
+          child: Row(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              SizedBox(
+                width: 120,
+                height: 40,
+                child: TextField(
+                  decoration: const InputDecoration(
+                    border: OutlineInputBorder(),
+                    hintText: 'Latitude',
+                  ),
+                  controller: _latitudeInput,
+                ),
+              ),
+              SizedBox(
+                width: 120,
+                height: 40,
+                child: TextField(
+                  decoration: const InputDecoration(
+                    border: OutlineInputBorder(),
+                    hintText: 'Longitude',
+                  ),
+                  controller: _longitudeInput,
+                ),
+              ),
+              SizedBox(
+                width: 120,
+                height: 40,
+                child: TextField(
+                  decoration: const InputDecoration(
+                    border: OutlineInputBorder(),
+                    hintText: 'Altitude',
+                  ),
+                  controller: _altitudeInput,
+                ),
+              ),
+            ],
+          ),
+        ),
+        const SizedBox(height: 16),
+        ElevatedButton(
```

Comment:
Try putting the last two `ElevatedButton`s in an `OverflowBar` widget with a set `spacing` attr